crates/db_views/post/src/test.rs

```diff
@@ -1255,6 +1256,330 @@ async fn pagination_includes_each_post_once(data: &mut Data) -> LemmyResult<()>
+    .await
+  };
+
+  let first_page = get_page(&None).await?;
+  assert_eq!(first_page.data.len(), page_size);
+  assert!(first_page.prev_page.is_none()); // without request cursor, no back cursor
+  assert!(first_page.next_page.is_some());
+
+  let last_page = get_page(&first_page.next_page).await?;
+  assert_eq!(last_page.data.len(), page_size);
+  assert!(last_page.prev_page.is_some());
+  assert!(last_page.next_page.is_some()); // full page, has cursor
+
+  // Get the first page with both cursors
+  let first_page2 = get_page(&last_page.prev_page).await?;
+  assert_eq!(first_page.data.len(), page_size);
+  assert!(first_page2.prev_page.is_some()); // full page, has cursor
+  assert!(first_page2.next_page.is_some());
+  assert_eq!(first_page2.next_page, first_page.next_page);
+  assert_eq!(
+    first_page2
+      .data
+      .into_iter()
+      .map(|pv| pv.post.id)
+      .collect::<Vec<PostId>>(),
+    first_page
+      .data
+      .clone()
+      .into_iter()
+      .map(|pv| pv.post.id)
+      .collect::<Vec<PostId>>()
+  );
+
+  let beyond_first_page = get_page(&first_page2.prev_page).await?;
+  assert_eq!(beyond_first_page.data.len(), 0);
+  assert!(beyond_first_page.prev_page.is_none());
+  assert!(beyond_first_page.next_page.is_some());
+
+  let recovered_first_page = get_page(&beyond_first_page.next_page).await?;
+  assert_eq!(recovered_first_page.data.len(), page_size);
+  assert!(recovered_first_page.prev_page.is_some()); // full page, has cursor
+  assert!(recovered_first_page.next_page.is_some());
+  assert_eq!(recovered_first_page.next_page, first_page2.next_page);
+  assert_eq!(recovered_first_page.prev_page, first_page2.prev_page);
+  assert_eq!(
+    recovered_first_page
+      .data
+      .into_iter()
+      .map(|pv| pv.post.id)
+      .collect::<Vec<PostId>>(),
+    first_page
+      .data
+      .into_iter()
+      .map(|pv| pv.post.id)
+      .collect::<Vec<PostId>>()
+  );
+
+  let beyond_last_page = get_page(&last_page.next_page).await?;
+  assert_eq!(beyond_last_page.data.len(), 0);
+  assert!(beyond_last_page.prev_page.is_some());
+  assert!(beyond_last_page.next_page.is_none());
+
+  let recovered_last_page = get_page(&beyond_last_page.prev_page).await?;
+  assert_eq!(recovered_last_page.data.len(), page_size);
+  assert!(recovered_last_page.prev_page.is_some());
+  assert!(recovered_last_page.next_page.is_some()); // full page, has cursor
+  assert_eq!(recovered_last_page.next_page, last_page.next_page);
+  assert_eq!(recovered_last_page.prev_page, last_page.prev_page);
+  assert_eq!(
+    recovered_last_page
+      .data
+      .into_iter()
+      .map(|pv| pv.post.id)
+      .collect::<Vec<PostId>>(),
+    last_page
+      .data
+      .into_iter()
+      .map(|pv| pv.post.id)
+      .collect::<Vec<PostId>>()
+  );
+
+  LocalUser::delete(pool, inserted_reader_local_user.id).await?;
+  Community::delete(pool, second_community.id).await?;
+  Community::delete(pool, first_community.id).await?;
```

Comment:
Its enough to call `Instance::delete` here for  data.instance.id, then everything created on that instance also gets deleted.